archive/tar.go

```diff
@@ -350,8 +350,11 @@ func createTarFile(ctx context.Context, path, extractDir string, hdr *tar.Header
 		}
 
 		_, err = copyBuffered(ctx, file, reader)
-		if err1 := file.Close(); err == nil {
-			err = err1
+
+		for _, op := range []func() error{file.Sync, file.Close} {
```

Comment:
I was working on this. I think we still need to sync parent dir to ensure that dir has metadata to the entry.

Reply:
I will share some test data in your pr.

Reply:
@fuweid 

> I was working on this. I think we still need to sync parent dir to ensure that dir has metadata to the entry.

Is there an update on this PR? Is there a performance impact? Very much hope this issue can be resolved.

Reply:
Hi @lijinyong ,still working in progress. I don't want to introduce sync-fs operation since it might increase IO pressure. Need more time to verify it.

Reply:
ping @lijinyong it has been fixed by https://github.com/containerd/containerd/pull/9401